c10/cuda/CUDAStream.h

```diff
@@ -67,7 +68,12 @@ class C10_CUDA_API CUDAStream {
   /// Construct a CUDAStream from a Stream.  This construction is checked,
   /// and will raise an error if the Stream is not, in fact, a CUDA stream.
   explicit CUDAStream(Stream stream) : stream_(stream) {
-    TORCH_CHECK(stream_.device_type() == DeviceType::CUDA);
+    TORCH_CHECK(
+        stream_.device_type() == DeviceType::CUDA,
+        "Got device type ",
+        stream_.device_type(),
+        "\n",
+        c10::get_backtrace());
```

Comment:
This was a change to help me debug some ROCm failures, I'll undo it before landing